# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/librsvg/hotfix-i486.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- librsvg-2.60.0/vendor/target-lexicon/src/targets.rs.vanilla	2025-05-29 19:17:18.231742517 +0200
+++ librsvg-2.60.0/vendor/target-lexicon/src/targets.rs	2025-05-29 19:45:19.803446319 +0200
@@ -467,6 +467,7 @@
 #[allow(missing_docs)]
 pub enum X86_32Architecture {
     I386,
+    I486,
     I586,
     I686,
 }
@@ -478,6 +479,7 @@
 
         match self {
             I386 => Cow::Borrowed("i386"),
+            I486 => Cow::Borrowed("i486"),
             I586 => Cow::Borrowed("i586"),
             I686 => Cow::Borrowed("i686"),
         }
@@ -1196,6 +1198,7 @@
 
         Ok(match s {
             "i386" => I386,
+            "i486" => I486,
             "i586" => I586,
             "i686" => I686,
             _ => return Err(()),
--- librsvg-2.60.0/vendor/cfg-expr/src/targets/builtins.rs.vanilla	2025-05-29 17:13:13.228964061 +0200
+++ librsvg-2.60.0/vendor/cfg-expr/src/targets/builtins.rs	2025-05-29 17:15:11.869323244 +0200
@@ -1223,6 +1223,19 @@
         panic: Panic::unwind,
     },
     TargetInfo {
+        triple: Triple::new_const("i486-unknown-linux-gnu"),
+        os: Some(Os::linux),
+        abi: None,
+        arch: Arch::x86,
+        env: Some(Env::gnu),
+        vendor: Some(Vendor::unknown),
+        families: Families::unix,
+        pointer_width: 32,
+        endian: Endian::little,
+        has_atomics: HasAtomics::atomic_8_16_32_ptr,
+        panic: Panic::unwind,
+    },
+    TargetInfo {
         triple: Triple::new_const("i586-unknown-linux-gnu"),
         os: Some(Os::linux),
         abi: None,
@@ -1236,6 +1249,19 @@
         panic: Panic::unwind,
     },
     TargetInfo {
+        triple: Triple::new_const("i486-unknown-linux-musl"),
+        os: Some(Os::linux),
+        abi: None,
+        arch: Arch::x86,
+        env: Some(Env::musl),
+        vendor: Some(Vendor::unknown),
+        families: Families::unix,
+        pointer_width: 32,
+        endian: Endian::little,
+        has_atomics: HasAtomics::atomic_8_16_32_ptr,
+        panic: Panic::unwind,
+    },
+    TargetInfo {
         triple: Triple::new_const("i586-unknown-linux-musl"),
         os: Some(Os::linux),
         abi: None,
